virtool/analyses/api.py

```diff
@@ -242,3 +245,39 @@ async def blast(req: aiohttp.web.Request) -> aiohttp.web.Response:
     }
 
     return json_response(blast_data, headers=headers, status=201)
+
+
+@routes.patch("/api/analyses/{analysis_id}", schema={
+    "results": {"type": "dict", "required": True}
+})
+async def patch_analysis(req: aiohttp.web.Request):
+    """Sets the result for an analysis and marks it as ready."""
+    db: DB = req.app["db"]
+    analyses: Collection = db.analyses
+    analysis_id: str = req.match_info["analysis_id"]
+
+    analysis_document: Dict[str, Any] = await analyses.find_one({"_id": analysis_id})
+
+    if not analysis_document:
+        return not_found(f"There is no analysis with id {analysis_id}")
+
+    if "ready" in analysis_document and analysis_document["ready"]:
+        return conflict("There is already a result for this analysis.")
+
+    request_json = await req.json()
+
+    await analyses.update_one({"_id": analysis_id}, {
+        "$set": {
+            "results": request_json["results"],
+            "ready": True
+        }
+    })
+
+    await recalculate_workflow_tags(db, analysis_document["sample"]["id"])
+
+    return aiohttp.web.Response(status=200)
```

Comment:
Return the state of the analysis after it has been changed.

```python
document = await analyses.find_one_and_update({"_id": analysis_id}, {
    "$set": {
        "results": request_json["results"],
        "ready": True
    }
})

await recalculate_workflow_tags(db, analysis_document["sample"]["id"])

return json_response(base_processor(document))
```

`base_processor` changes `_id` to `id`.